src/bp/nlu-core/utterance/utterance.ts

```diff
@@ -137,7 +148,14 @@ export default class Utterance {
 
       const weight = Math.min(1, token.tfidf) // TODO: there's already an upper limit on TFIDF
       totalWeight += weight
-      const weightedVec = scalarMultiply(token.vector as number[], weight / norm) // TODO: experiment without dividing by norm
+
+      /**
+       * TODO: experiment without dividing by norm
```

Comment:
not of fan of keeping todos in the codebase ... I I know it was already there but can you keep it for you  ..

Reply:
I don't know... I feel like scientific code and mathematical equations need some sort of documentation. This is not a self explenatory algorithm and I don't want the reader (most of the time, the reader is me) to feel like an algorithm is arbitrary or simply a mistake.

If not, I end up asking myself way too much questions when reading the NLU code; questions I sometimes have already asked myself in the past. 

This precise comment, is not actually a TODO, but more of a self note that we could **not divide by norm** and we've made the engineering choice of **dividing by norm**.

This TODO comment could become something like:

```ts
/*
* The follwing posts suggests there might important information in a word vector's length:
* - https://stats.stackexchange.com/questions/177905/should-i-normalize-word2vecs-word-vectors-before-using-them
* - https://stackoverflow.com/questions/36034454/what-meaning-does-the-length-of-a-word2vec-vector-have
*/
```

Reply:
Much better

Reply:
:+1: